SourceX/sdl_ptrs.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+/**
+ * @brief std::unique_ptr specializations for SDL types.
+ */
+
+#include <memory>
+#include <type_traits>
+
+#include <SDL.h>
+
+namespace dvl {
+
+/**
+ * @brief Deletes the SDL surface using `SDL_FreeSurface`.
+ */
+struct SDLSurfaceDeleter {
+	void operator()(SDL_Surface *surface) const
+	{
+		SDL_FreeSurface(surface);
+	}
+};
+
+using SDLSurfaceUniquePtr = std::unique_ptr<SDL_Surface, SDLSurfaceDeleter>;
+
+/**
+ * @brief Deletes the object using `SDL_free`.
+ */
+template <typename T>
+struct SDLFreeDeleter {
+	static_assert(!std::is_same<T, SDL_Surface>::value,
+	    "SDL_Surface should use SDLSurfaceUniquePtr instead.");
+
+	void operator()(T *obj) const

```

Comment:
Is this really a `const` method? Doesn't freeing something cause side effects?

Reply:
Yes, it's a const method because it does not modify any of the struct's fields (it doesn't have any)